### PR TITLE
Add routing in search autocomplete component

### PR DESCRIPTION
**Pull request by bazelajoanna**
Add routing in search autocomplete component to enable opening problem from the suggestion list (using keyboard and clicking).

### CODE REVIEW

File: src/App.test.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import { screen, within } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import { testRenderer } from "./setupTests";
+import App from "./App";
+
+describe("App", () => {
```
Comment (franiakat) [inline src/App.test.tsx]:
```suggestion
describe("<App />", () => {
```

File: src/App.test.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  it("navigates to 'O metodzie' page after clicking on it on header", async () => {
```
Comment (franiakat) [inline src/App.test.tsx]:
```suggestion
  it("navigates to 'About' page after clicking on it on header", async () => {
```

File: src/App.test.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    testRenderer(<App />);
+    const aboutPage = screen.getByText(/O metodzie/i);
+    userEvent.click(aboutPage);
+
+    expect(screen.getByText("Co to są Myślące Klasy?")).toBeInTheDocument();
+  });
+
+  it("navigates to 'Strefa moderatora' page after clicking on it on header", async () => {
```
Comment (franiakat) [inline src/App.test.tsx]:
```suggestion
  it("navigates to 'Moderator' page after clicking on it on header", async () => {
```

File: src/App.test.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import userEvent from "@testing-library/user-event";
+
+import { testRenderer } from "./setupTests";
+import App from "./App";
+
+describe("App", () => {
+  it("navigates to 'O metodzie' page after clicking on it on header", async () => {
+    testRenderer(<App />);
+    const aboutPage = screen.getByText(/O metodzie/i);
+    userEvent.click(aboutPage);
```
Comment (franiakat) [inline src/App.test.tsx]:
```suggestion
    const aboutPageNavLink = screen.getByText(/O metodzie/i);
    userEvent.click(aboutPageNavLink);
```

File: src/App.test.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+    expect(screen.getByText("Co to są Myślące Klasy?")).toBeInTheDocument();
+  });
+
+  it("navigates to 'Strefa moderatora' page after clicking on it on header", async () => {
+    testRenderer(<App />);
+    const moderatorPage = screen.getByText(/Strefa moderatora/i);
+    userEvent.click(moderatorPage);
```
Comment (franiakat) [inline src/App.test.tsx]:
```suggestion
    const moderatorPageNavLink = screen.getByText(/Strefa moderatora/i);
    userEvent.click(moderatorPageNavLink);
```

File: src/App.test.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+    expect(screen.getByText("Witaj, moderatorze!")).toBeInTheDocument();
+  });
+
+  it("navigates to main page after clicking on the logo", async () => {
+    testRenderer(<App />);
+    const moderatorPage = screen.getByText(/Strefa moderatora/i);
+    userEvent.click(moderatorPage);
```
Comment (franiakat) [inline src/App.test.tsx]:
```suggestion
    const moderatorPageNavLink = screen.getByText(/Strefa moderatora/i);
    userEvent.click(moderatorPageNavLink);
```

**Review by franiakat — CHANGES_REQUESTED** [review]
Very nice! 🎉 Just a few adjustments propositions from my side, nothing major.

**Review by evulins — APPROVED** [review]
Beautiful! Works like a charm :)

**Review by franiakat — APPROVED** [review]
AWESOME, ready to be:
sqash

**Pull request merged**